packages/circuit-ui/components/Headline/Headline.docs.mdx

```diff
@@ -4,25 +4,118 @@ import { Status, Props, Story } from '../../../../.storybook/components';
+different sections.
 
-Skipping heading ranks can be confusing and should be avoided where possible: Ensure that an `h2` is not followed directly by an `h4`, for example. It is ok to skip ranks when closing subsections, for instance, an `h2` beginning a new section can follow an `h4` as it closes the previous section.
+### Best practices
 
-[Learn more about a proper page structure](https://www.w3.org/WAI/tutorials/page-structure/headings/).
+#### Each page should have a single `<h1>`
 
-## Component variations
+The `<h1>` represents the page title. Our pages should all have one (and only
+one), and it should be similar to the document's `<title>`.
 
-### Sizes
+#### Headings should be descriptive
 
-The Headline component comes in four sizes.
+A heading's text should describe its section's content accurately and concisely.
+Because screen reader users will use them to navigate the page's contents,
+headings must also make sense out of the content's context, as if rendered in a
+table of contents.
 
-Use the `four` size for card headers and `three` for page titles in web applications. For specific use cases such as landing pages, consider using the [Title](Typography/Title) component.
+#### Headings should be nested incrementally
 
-<Story id="typography-headline--sizes" />
+Beyond the `<h1>`, heading levels should be nested incrementally to make them
+easily accessible to users. Think of your heading structure like a nested list:
+
+- `<h1>` Page title
+  - `<h2>` Section title
+    - `<h3>` Section title
+    - `<h3>` Section title
+      - `<h4>` Section title
+  - `<h2>` Section title
+
+Don't skip heading levels: an `<h4>` should not directly follow an `<h2>`.
+
+In contrast, skipping headline _sizes_ and jumping from e.g.
+`<Headline size="two" as="h2" />` to `<Headline size="four" as="h4" />` does not
+constitute an accessibility issue, although we recommend keeping sizes visually
+consistent across pages.
+
+#### Avoid sections without headings
+
+If a section doesn't have a heading (for example, if it uses a different
+background color to distinguish it from another section), it will not be easily
+accessible to blind users.
+
+Consider adding a heading to make the section's purpose clearer.
+
+You can also add visually hidden headings using the `hideVisually()` style
+mixin:
+
+```tsx
+import { Headline, hideVisually } from '@sumup/circuit-ui';
+
+function HiddenHeadline() {
+  return (
+    <Headline as="h2" css={hideVisually}>
+      My section
+    </Headline>
+  );
+}
+```
+
+#### Avoid headings without sections
+
+Headings typically use large typography, but all large typography shoulnd't be a
+heading.
+
+If you need to render large text that doesn't describe a section, don't use a
+semantic heading element, and instead use CSS to enlarge the text.
+
+```tsx
+<section>
+  <Headline as="h2" size="four">
+    My product
+  </Headline>
+  <Headline as="p" size="one">
```

Comment:
We don't support this right now. Should we? Is there a reason not to? If we won't support this, how do we cater to the use case of a developer wanting to use large typography without a semantic heading element?

Reply:
cc @connor-baer—thoughts?

Reply:
While there is no strict coupling between size and heading level, they should be roughly aligned. Larger text communicates importance to sighted users and thus should also be reflected semantically for visually-impaired users.

I can only think of a single use case for large text that's not a headline: the price tags on the marketing website. IMO, a custom styled component makes more sense there. **Edit**: I just realized that you've used a price as the example here 😃

Reply:
Sounds good, I'll document this. Maybe we can recommend using the typography values from the theme